control_panel_api/authentication.py

```diff
@@ -60,15 +60,38 @@ def get_or_create_user(decoded_payload):
     try:
         user = User.objects.get(pk=decoded_payload.get('sub'))
     except User.DoesNotExist:
-        user = User.objects.create(
-            pk=decoded_payload.get('sub'),
-            username=decoded_payload.get(settings.OIDC_FIELD_USERNAME),
-            email=decoded_payload.get(settings.OIDC_FIELD_EMAIL),
-            name=decoded_payload.get(settings.OIDC_FIELD_NAME),
-        )
-        user.save()
+        pass
+    logger.info('User "{}" is new. Will add it to the db, create aws role and '
+                'create its charts on k8s.'.format(decoded_payload.get('sub')))
+    user = User.objects.create(
+        pk=decoded_payload.get('sub'),
+        username=decoded_payload.get(settings.OIDC_FIELD_USERNAME),
+        email=decoded_payload.get(settings.OIDC_FIELD_EMAIL),
+        name=decoded_payload.get(settings.OIDC_FIELD_NAME),
+    )
+    user.save()
+    logger.info('User {} saved to db as: "{}"'
+                .format(decoded_payload.get('sub'), user))
+    try:
         user.aws_create_role()
         user.helm_create()
+    except Exception:
```

Comment:
I'm not a great fan of `except Exception` as it's very generic and usually is a bad idea.

I see that we're kind of rolling back the creation of the user if there was an exception, a couple of thoughts on this:
- We're assuming the user wasn't there (bear in mind the method is called `get_or_create_user()`), but what if it was? e.g. a random exception occurred, and then my user is deleted which is not good.
- IAM role creation and helm charts installation were previously done only when the user was created, this behaviour now changed and this could cause more exceptions (e.g. because trying to create an IAM role which is already there)
- As we only rollback the user creation (or at least that's the intent) maybe this should be instead wrapped in a DB transaction so that the DB/django takes care of it. Revert changed to AWS/helm could be more complicated and probably not worth.

Reply:
I like the idea to use a db transaction to roll-back the creation of the user